src/serialize.rs

```diff
@@ -323,20 +367,38 @@ impl EGraph {
                 .node_ids
                 .entry(class_id.clone())
                 .or_insert_with(|| {
-                    // If we don't find node IDs for this class, it means that all nodes for it were omitted due to size constraints
-                    // In this case, add a dummy node in this class to represent the missing nodes
-                    let node_id = self.to_node_id(Some(sort), SerializedNode::Dummy(value));
-                    serializer.result.nodes.insert(
-                        node_id.clone(),
-                        egraph_serialize::Node {
-                            op: "[...]".to_string(),
-                            eclass: class_id.clone(),
-                            cost: NotNan::new(f64::INFINITY).unwrap(),
-                            children: vec![],
-                            subsumed: false,
-                        },
-                    );
-                    VecDeque::from(vec![node_id])
+                    // Check if this e-class is empty (never had outputs) or omitted (had outputs but were truncated/discarded)
+                    let is_empty = !serializer.eclasses_with_outputs.contains(class_id);
+                    if is_empty {
+                        // Empty e-class: use empty string as the name
+                        serializer.empty_eclasses.push(class_id.to_string());
+                        let node_id = self.to_node_id(Some(sort), SerializedNode::Dummy(value));
+                        serializer.result.nodes.insert(
+                            node_id.clone(),
+                            egraph_serialize::Node {
+                                op: "".to_string(),
```

Comment:
I think it might be helpful to be more explicit here. Like EMPTY-ECLASS or something.